### PR TITLE
docs: update ts docs

### PR DESCRIPTION
**Pull request by liuyang1520**
# Amplitude Developer Docs PR
## Description
This PR updates the TS SDKs docs:
- how to override sdk configuration with ampli load
- add custom http client guidance section
- async flow control of flush and track

## Deadline

When do these changes need to be live on the site?

## Change type

- [x] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [x] My documentation follows the style guidelines of this project.
- [x] I previewed my documentation on a local server using `mkdocs serve`.
- [x] Running `mkdocs serve` didn't generate any failures.
- [x] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

File: docs/data/sdks/typescript-browser/ampli.md [hunk 1]
```diff
@@ -97,6 +97,34 @@ The `load()` function accepts an options object to configure the SDK's behavior:
 | `client.instance`                     | Optional. `AmplitudeClient`. Specifies an Amplitude instance. By default Ampli creates an instance for you.|
 | `client.configuration`                | Optional. `Amplitude.Config`. Overrides the default configuration for the AmplitudeClient.|
 
+Example of initialization with `load` to override the default configuration:
+
+=== "TypeScript"
+
+    ```js
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/ampli.md:104]: ts?

File: docs/data/sdks/typescript-browser/index.md [hunk 2]
```diff
@@ -274,6 +274,14 @@ The `flush` method triggers the client to send buffered events immediately.
 flush();
 ```
 
+By default, `flush` is called automatically in an interval, if you want to flush the events all together, you can control the async flow with the optional Promise interface, example:
+
+```typescript
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-browser/index.md:279]: ts?

File: docs/data/sdks/typescript-node/ampli.md [hunk 3]
```diff
@@ -95,6 +95,34 @@ The `load()` function accepts an options object to configure the SDK's behavior:
 | `client.instance`                     | Optional. `AmpltitudeClient`. Specifies an Amplitude instance. By default Ampli creates an instance for you.|
 | `client.configuration`                | Optional. `Amplitude.Config`. Overrides the default configuration for the AmplitudeClient.|
 
+Example of initialization with `load` to override the default configuration:
+
+=== "TypeScript"
+
+    ```js
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-node/ampli.md:102]: ts?

File: docs/data/sdks/typescript-react-native/ampli.md [hunk 4]
```diff
@@ -93,6 +93,34 @@ The `load()` function accepts an options object to configure the SDK's behavior:
 | `client.instance`                     | Optional. `AmpltitudeClient`. Specifies an Amplitude instance. By default Ampli creates an instance for you.|
 | `client.configuration`                | Optional. `Amplitude.Config`. Overrides the default configuration for the AmplitudeClient.|
 
+Example of initialization with `load` to override the default configuration:
+
+=== "TypeScript"
+
+    ```js
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-react-native/ampli.md:100]: ts

File: docs/data/sdks/typescript-react-native/index.md [hunk 5]
```diff
@@ -311,6 +311,14 @@ import { flush } from '@amplitude/analytics-react-native';
 flush();
 ```
 
+By default, `flush` is called automatically in an interval, if you want to flush the events all together, you can control the async flow with the optional Promise interface, example:
+
+```typescript
```
Comment (yuhao900914) [inline docs/data/sdks/typescript-react-native/index.md:316]: ts

**Review by yuhao900914 — COMMENTED** [review]
Feels there are too many identical things, since we didn’t have the ‘import’, should we use the shared doc?

File: docs/data/sdks/typescript-react-native/index.md (hunk 5 above)
Comment (liuyang1520) [reply docs/data/sdks/typescript-react-native/index.md:316]: Thanks for reviewing! I tested locally, think `ts` and `typescript` are the same, I follow the same convention in the same file.

**Comment by liuyang1520** [comment]
> Feels there are too many identical things, since we didn’t have the ‘import’, should we use the shared doc?

I feel our doc is a little bit bloated too. The browser/rn one and node one have slightly difference in the `init` and configuration options. Any suggestions?

**Review by caseyamp — APPROVED** [review]

File: docs/data/sdks/typescript-react-native/index.md (hunk 5 above)
Comment (kevinpagtakhan) [reply docs/data/sdks/typescript-react-native/index.md:316]: yes either is fine

**Review by yuhao900914 — APPROVED** [review]

**Pull request merged**